### DISCUSSION TITLE
Avoid NullPointerExceltions by using a getter for wrapped values with @Nullable annotation

### DISCUSSION THREAD

**Discussion by HenrikJannsen** [bisq-network/bisq2#835] [General]
We should use always the `@Nullable` annotation if a value can be null.
When using JavaFX properties or other swappers (Observeable) we cannot apply the `@Nullable` annotation as the wrapper is final and never null though the wrapped value can be null. 
To avoid running into NullPointerExceltions we should use a getter method when accessing the value and annotate the getter method with `@Nullable`. 

In the UI code we use direct field access as the model is static private and accessible. We should use getter methods instead to gain more control.
In case we have a nullable value we should postfix the property field with  `Property` to make it more clearly distinguished from the getter.

```
        private final ObjectProperty<ChatChannel<? extends ChatMessage>> channelProperty = new SimpleObjectProperty<>();
        @Nullable
        public ChatChannel<? extends ChatMessage> getChannel() {
            return channelProperty.get();
        }
```

**Comment by DrColver** [comment]
Using getters, and setters, would be good. I added a setter to `UserIdentityStore` to ensure the correct usage of the selectedUserIdentity `Observable`. If the `Property` postfix scheme is to be used, then it would follow that `Observable` should be used as a postfix in corresponding situations.

**Reply by HenrikJannsen** [reply]
Yes probably best to follow the JavaFx property conventions.

**Comment by HenrikJannsen** [comment]
Not so sure anymore about that. When applying it to some new classes it did not felt like an improvement as the model classes get very boilerplate. Lombok helps to keep them minimal... Will need more time to get a clear opinion.

**Reply by DrColver** [reply]
It's true that you get a lot more boilerplate code which hampers readability. As long as the conventions are used consistently either way is good. Preferably over the entire project.

**Comment by HenrikJannsen** [comment]
I guess we can close that. The extra pain with boilerplate does not justify the benefits IMO.
